internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -498,14 +487,184 @@ func resourceApiManagementApiCreateUpdate(d *pluginsdk.ResourceData, meta interf
+				return fmt.Errorf("creating/updating %s: %+v", *id, err)
+			}
+		}
+	}
+
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	if resp.Model == nil || resp.Model.Properties == nil {
+		return fmt.Errorf("retrieving %s: `properties` was nil", *id)
+	}
+
+	existing := resp.Model.Properties
+	prop := &api.ApiCreateOrUpdateProperties{
+		Path:                          existing.Path,
+		Protocols:                     existing.Protocols,
+		ServiceURL:                    existing.ServiceURL,
+		Description:                   existing.Description,
+		ApiVersionDescription:         existing.ApiVersionDescription,
+		ApiRevisionDescription:        existing.ApiRevisionDescription,
+		SubscriptionRequired:          existing.SubscriptionRequired,
+		SubscriptionKeyParameterNames: existing.SubscriptionKeyParameterNames,
+		AuthenticationSettings:        existing.AuthenticationSettings,
+		Contact:                       existing.Contact,
+		License:                       existing.License,
+		SourceApiId:                   existing.SourceApiId,
+		DisplayName:                   existing.DisplayName,
+		ApiVersion:                    existing.ApiVersion,
+		ApiVersionSetId:               existing.ApiVersionSetId,
+		TermsOfServiceURL:             existing.TermsOfServiceURL,
+		Type:                          existing.Type,
+	}
+
+	if d.HasChange("path") {
+		prop.Path = path
+	}
+
+	if d.HasChange("protocols") {
+		prop.Protocols = protocols
+	}
+
+	if d.HasChange("api_type") {
+		prop.Type = pointer.To(apiType)
+		prop.ApiType = pointer.To(soapApiType)
+	}
+
+	if d.HasChange("service_url") {
+		prop.ServiceURL = pointer.To(serviceUrl)
+	}
+
+	if d.HasChange("description") {
+		prop.Description = pointer.To(d.Get("description").(string))
+	}
+
+	if d.HasChange("revision_description") {
+		prop.ApiRevisionDescription = pointer.To(d.Get("revision_description").(string))
+	}
+
+	if d.HasChange("version_description") {
+		prop.ApiVersionDescription = pointer.To(d.Get("version_description").(string))
+	}
+	if d.HasChange("subscription_required") {
+		prop.SubscriptionRequired = pointer.To(d.Get("subscription_required").(bool))
+	}
+
+	if d.HasChange("subscription_key_parameter_names") {
+		subscriptionKeyParameterNamesRaw := d.Get("subscription_key_parameter_names").([]interface{})
+		prop.SubscriptionKeyParameterNames = expandApiManagementApiSubscriptionKeyParamNames(subscriptionKeyParameterNamesRaw)
+	}
+
+	authenticationSettings := &api.AuthenticationSettingsContract{}
+	if d.HasChange("oauth2_authorization") {
+		oAuth2AuthorizationSettingsRaw := d.Get("oauth2_authorization").([]interface{})
+		oAuth2AuthorizationSettings := expandApiManagementOAuth2AuthenticationSettingsContract(oAuth2AuthorizationSettingsRaw)
+		authenticationSettings.OAuth2 = oAuth2AuthorizationSettings
+		prop.AuthenticationSettings = authenticationSettings
```

Comment:
The way this is written will reset `openid_authentication` if it doesn't change as it's setting `oauth2_authorization` into an empty `AuthenticationSettingsContract`

We should modify the existing `AuthenticationSettings` rather than creating a new one